src/runtime/hexagon_remote/halide_remote.cpp

```diff
@@ -231,8 +242,19 @@ int halide_hexagon_remote_run(handle_t module_ptr, handle_t function,
         *next_arg = input_scalarsPtrs[i].data;
     }
 
+    // Prior to running the pipeline, power HVX on (if it was not already on).
+    int result = halide_hexagon_remote_power_hvx_on();
+    if (result != 0) {
+        return result;
+    }
+
     // Call the pipeline and return the result.
-    return run_context.run(pipeline, args);
+    result = run_context.run(pipeline, args);
+
+    // Power HVX off.
+    halide_hexagon_remote_power_hvx_off();
```

Comment:
Should we instead power on and power off at a level about halide_hexagon_remote_run? Otherwise, unless i am missing something, it appears to me that multiple small calls to halide_hexagon_remote_run will switch on and off HVX every single time, which is not a bad thign if the time between such kernels is substantial. But the trade off is worth considering.